File: packages/health_connector_hk_ios/lib/src/mappers/measurement_unit_mappers/mass_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Mass, sinceV1_0_0;
import 'package:health_connector_hk_ios/src/pigeon/health_connector_hk_ios_api.g.dart'
    show MassDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Mass] to [MassDto].
@sinceV1_0_0
@internal
/// Converts [Mass] to [MassDto].
@sinceV1_0_0
@internal
extension MassToDto on Mass {
  MassDto toDto() {
    // Uses kilograms as the transfer unit for consistency.
    return MassDto(kilograms: inKilograms);
  }
}

/// Converts [MassDto] to [Mass].
@sinceV1_0_0
@internal
extension MassDtoToDomain on MassDto {
  Mass toDomain() {
    return Mass.kilograms(kilograms);
  }
}

File: packages/health_connector_hk_ios/ios/health_connector_hk_ios/Sources/health_connector_hk_ios/mappers/measurement_unit_mappers/MassMapper.swift
import Foundation
import HealthKit

extension MassDto {
    /// Converts this DTO to a HealthKit `HKQuantity`.
    func toHealthKit() -> HKQuantity {
        HKQuantity(unit: .gramUnit(with: .kilo), doubleValue: kilograms)
    }
}

extension HKQuantity {
    /// Converts this HealthKit quantity to a `MassDto`.
    ///
    /// Uses kilograms as the transfer unit for consistency.
    func toMassDto() -> MassDto {
        let kilograms = doubleValue(for: .gramUnit(with: .kilo))
        return MassDto(kilograms: kilograms)
    }
}
